### PR TITLE
(feat): done,undone,clear,session

### PR DESCRIPTION
**Pull request by harsh-vador**

### CODE REVIEW

File: .eslintcache [hunk 1]
```diff
@@ -1 +1 @@
...
...sName={todo.isComplete ? 'todo-row complete' : 'todo-row'} key={index}>\n            <input type=\"checkbox\" onChange={()=> {completeTodo(todo.id)}}/>\n            <div key={todo.id} onClick={() => completeTodo(todo.id)}>\n                {todo.text}\n            </div>\n            <div className='icons'>\n                <RiCloseCircleLine\n                    onClick={() => removeTodo(todo.id)}\n                    className='delete-icon'\n                />\n                \n                \n            </div>\n        </div>\n    ));\n};\n\nexport default Todo;\n","/Users/deuex/Desktop/todo/to-do/react-todo-list-master/src/components/TodoForm.js",[],{"ruleId":"28","replacedBy":"29"},{"ruleId":"30","replacedBy":"31"},{"ruleId":"32","severity":1,"message":"33","line":1,"column":17,"nodeType":"34","messageId":"35","endLine":1,"endColumn":25},{"ruleId":"32","severity":1,"message":"36","line":2,"column":8,"nodeType":"34","messageId":"35","endLine":2,"endColumn":16},"no-native-reassign",["37"],"no-negated-in-lhs",["38"],"no-unused-vars","'useState' is defined but never used.","Identifier","unusedVar","'TodoList' is defined but never used.","no-global-assign","no-unsafe-negation"]
```
Comment (shah-harshit) [inline .eslintcache]: We don't need to commit generated files. Please add such files to .gitignore and avoid committing them.

File: src/components/TodoList.js [hunk 2]
```diff
@@ -45,13 +45,21 @@ function TodoList() {
             return todo;
         })
         setTodos(updatedTodos);
+        console.log(todos)
```
Comment (shah-harshit) [inline src/components/TodoList.js]: Remove all the console log's from this file.

File: src/components/TodoList.js [hunk 3]
```diff
@@ -45,13 +45,21 @@ function TodoList() {
...
 
+        setTodos(removeArr);
+    }
+    
     return (
         <div className="todo-context">
             <h1>To-Do List</h1>
             <TodoForm onSubmit={addTodo} />
             <Todo todos={todos} completeTodo={completeTodo} removeTodo={removeTodo}  />
+            <button className="todo-button" onClick={deleteAll}>Clear All</button>
```
Comment (shah-harshit) [inline src/components/TodoList.js]: The button text says "Clear All" and the function is named "deleteAll". Please make it consistent.

File: src/components/TodoList.js [hunk 4]
```diff
@@ -45,13 +45,21 @@ function TodoList() {
             return todo;
         })
         setTodos(updatedTodos);
+        console.log(todos)
     };
+    const deleteAll = id =>{
+        console.log(id)
+        const removeArr = [...todos].filter(todo => console.log(todo));
```
Comment (shah-harshit) [inline src/components/TodoList.js]: Looks like this filter doesn't do anything. Please review the logic.

File: src/components/Todo.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
+import TodoList from './TodoList';
 import { RiCloseCircleLine } from 'react-icons/ri';
 
-function Todo({ todos, completeTodo, removeTodo, updateTodo }) {
+function Todo({ todos, completeTodo, removeTodo }) {
  
 
     return todos.map((todo, index) => (
         <div className={todo.isComplete ? 'todo-row complete' : 'todo-row'} key={index}>
+            <input type="checkbox" onChange={()=> {completeTodo(todo.id)}}/>
```
Comment (shah-harshit) [inline src/components/Todo.js]: Create a function and pass it to onChange and onClick.

File: .eslintcache (hunk 1 above)
Comment (harsh-vador) [reply .eslintcache]: okay got it

File: src/components/TodoList.js (hunk 2 above)
Comment (harsh-vador) [reply src/components/TodoList.js]: Done

File: src/components/TodoList.js (hunk 3 above)
Comment (harsh-vador) [reply src/components/TodoList.js]: Changes made

File: src/components/TodoList.js (hunk 4 above)
Comment (harsh-vador) [reply src/components/TodoList.js]: Made the changes

File: src/components/Todo.js (hunk 5 above)
Comment (harsh-vador) [reply src/components/Todo.js]: DONE

**Review by shah-harshit — APPROVED** [review]

**Pull request merged**